src/dune/docker.py

```diff
@@ -66,21 +70,58 @@ def get_container(self):
     def get_image(self):
         return self._image
 
-    def execute_docker_cmd(self, cmd):
-        with subprocess.Popen(['docker'] + cmd,
-                              stdout=subprocess.PIPE, stderr=subprocess.PIPE) as proc:
-            stdout, stderr = proc.communicate()
-            if self._cl_args.debug:
-                print('docker '+' '.join(cmd))
-                print(stdout.decode('UTF-8'))
-                print(stderr.decode('UTF-8'))
-        return [stdout.decode('UTF-8'), stderr.decode('UTF-8'), proc.poll()]
+    @staticmethod
+    def print_streams(stdout, stderr):
+        if stdout is None and stderr is None:
+            print('!! No stdout/stderr info captured...')
+            return
+
+        print('================ STDOUT ================')
+        print(stdout)
+        print('================ STDERR ================')
+        print(stderr)
+        print('========================================')
+
+    def execute_docker_cmd(self, cmd, *, check_status=True, capture_output=True):
+        """Execute the given docker command in the active container.
+
+        :param check_status: if command has a return code != 0 then raise an exception
+        :param capture_output: whether to capture stdout and stderr and return them or to
+                               print the streams normally
+        :return: (stdout, stderr, status_code) if captured, (None, None, status_code) otherwise
+        """
+        if capture_output:
+            with subprocess.Popen(['docker'] + cmd,
+                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE) as proc:
+                stdout, stderr = proc.communicate()
+                stdout = stdout.decode('UTF-8')
+                stderr = stderr.decode('UTF-8')
+                status = proc.returncode
+
+        else:
+            with subprocess.Popen(['docker'] + cmd) as proc:
+                proc.communicate()
+                stdout = None
+                stderr = None
+                status = proc.returncode
+
+        if check_status and status != 0:
+            # some error happened, log it and fail
+            print(f'ERROR: docker {cmd}  -- returned: {status}')
+            self.print_streams(stdout, stderr)
+            raise docker_error
```

Comment:
What scenario did you encounter so you have decided to raise an exception?
Currently the error looks like this:
```
mikel@msi:~/repo/DUNE$ ./dune -- cleos wallet unlock
password: 
Error 3120005: Invalid wallet password
Are you sure you are using the right password?
Error Details:
Invalid password for wallet: "/root/eosio-wallet/./default.wallet"
ERROR: docker ['container', 'exec', '-i', 'dune_container', 'cleos', 'wallet', 'unlock']  -- returned: 1
!! No stdout/stderr info captured...
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/mikel/repo/DUNE/src/dune/__main__.py", line 84, in <module>
    dune_sys.execute_interactive_cmd(parser.get_forwarded_args())
  File "/home/mikel/repo/DUNE/src/dune/dune.py", line 478, in execute_interactive_cmd
    self._docker.execute_interactive_cmd(args)
  File "/home/mikel/repo/DUNE/src/dune/docker.py", line 181, in execute_interactive_cmd
    self.execute_cmd(cmd, interactive=True,
  File "/home/mikel/repo/DUNE/src/dune/docker.py", line 178, in execute_cmd
    return self.execute_docker_cmd(docker_cmd + cmd, **kwargs)
  File "/home/mikel/repo/DUNE/src/dune/docker.py", line 112, in execute_docker_cmd
    raise docker_error
docker.docker_error
```
and previously it was:
```
mikel@msi:~/repo/DUNE$ ./dune -- cleos wallet unlock
password: 
Error 3120005: Invalid wallet password
Are you sure you are using the right password?
Error Details:
Invalid password for wallet: "/root/eosio-wallet/./default.wallet"
```
The previous version is shorter and user should see error in the last line. I don't think user needs full stack trace.

Reply:
my bad, I wasn't properly catching the exception... I did it in another branch but it slipped through the cracks when doing this PR. I fixed it now. The motivation is that I'm using `dune` in scripts and I need it to return a non-zero code when it fails. Before it was printing the error but returning normally.

Before:
```
wackou@brannigan:/tmp/DUNE ‹main›$ ./dune -- cleos wallet unlock 
password: 
Error 3120005: Invalid wallet password
Are you sure you are using the right password?
Error Details:
Invalid password for wallet: "/root/eosio-wallet/./default.wallet"

wackou@brannigan:/tmp/DUNE ‹main›$ echo $?
0

```

After:
```
wackou@brannigan:~/dev/DUNE ‹docker_execute_cmd_rework*›$ ./dune -- cleos wallet unlock
password: 
Error 3120005: Invalid wallet password
Are you sure you are using the right password?
Error Details:
Invalid password for wallet: "/root/eosio-wallet/./default.wallet"
ERROR: docker ['container', 'exec', '-i', 'dune_container', 'cleos', 'wallet', 'unlock']  -- returned: 1
Docker Error

wackou@brannigan:~/dev/DUNE ‹docker_execute_cmd_rework*›$ echo $?
1
```